encoding/codecs/h264.md

```diff
@@ -0,0 +1,247 @@
+
+- ultrafast
+- superfast
+- veryfast
+- faster
+- fast
+- medium
+- slow
+- slower
+- veryslow
+- placebo
+
+You will almost never want to use the very extreme settings,
+but generally, if you want good quality
+and don't care about how long the encode takes,
+`slower` or `veryslow` are recommended.
+In this example, because we are just demonstrating how x264 works,
+we want a fast encode and have chosen `veryfast`.
+
+For the curious, you can see a full list of the settings
+enabled by each preset by running `x264 --fullhelp | less`.
+However, this probably won't mean much at the moment.
+Don't worry, this guide will explain later what all of those settings mean.
+
+#### `--tune animation`
+
+Beyond the preset chosen,
+x264 allows us to further tune the encoding settings
+for the type of content we're working with.
+The following tunings are generally the most useful:
+
+- `film`: Recommended for live action videos.
+- `animation`: Recommended for anime or cartoons with flat textures.
+For 3D animation (e.g. modern Pixar movies),
+you may find better results with `film`.
+- `grain`: Recommended for particularly grainy films.
+
+You don't need to use a tuning,
+but it generally helps to produce a better-looking video.
+
+#### `--crf 24`
+
+CRF is a constant-quality, 1-pass encoding mode.
+In layman's terms,
+this means that we don't need the output to meet a specific filesize,
+we just want the output to meet a certain quality level.
+CRF ranges from 0 to 51,
+with 0 being the best quality and 51 being the smallest filesize,
+but there is a certain range of CRF settings
+that are generally most useful.
+Here are some guidelines:
+
+- CRF 13: This is considered visually lossless to videophiles.
+This can produce rather large files,
+but is a good choice if you want high quality videos.
+Fansubbing groups often use this for Blu-ray encodes.
+- CRF 16-18: This is considered visually lossless to most viewers,
+and leans toward high quality while still providing a reasonable filesize.
+This is a typical range for fansub encodes.
+- CRF 21-24: This provides a good balance between quality and filesize.
+Some quality loss becomes visible, but this is generally a good choice
+where filesize becomes a concern, such as for videos viewed over the internet.
+- CRF 26-30: This prioritizes filesize, and quality loss becomes more obvious.
+It is generally not recommended to go higher than CRF 30 in any real-world
+encoding scenario, unless you want your videos to look like they were made
+for dial-up.
+
+#### `-o x264output.mkv -`
+
+This last portion tells which files to use for the input and output.
+We use `-o` to tell which filename to write the encoded file to.
+In this case, x264 will write a file at `x264output.mkv`
+in the current directory.
+
+The last argument to x264 is always the input file.
+In this case, we pass `-` for the input file,
+which tells x264 to use the piped output from vspipe.
+
+## Example 2
```

Comment:
Having a short title for what the examples are about helps finding the correct one later and gives a rough summary about their purpose.